### PR TITLE
Implement all missing OpenAL functions

### PR DESCRIPTION
**Pull request by yoanlcq**
This is of primary interest to projects that need to link against the entire OpenAL API (e.g bindings, and, in my particular case, [https://github.com/jpernst/alto/issues/30](https://github.com/jpernst/alto/issues/30)).
The PR also adds a minor fix in a test, and improves the existing `alGetEnumValue` implementation.

Tests as run with `pyhton tests/runner.py interactive.test_openal_*` passed.  I ran no tests other than these because all I did was add new functions.  
I did not create new tests either, since most of said functions are mere wrappers, and as such, their correctness is easy to prove with a review (IMO).  
  
Finally, I hope the commits are OK - I've tried a few things to somehow turn them all into a single one, but it messed up things every time, and I still have no clue.

### CODE REVIEW

File: src/library_openal.js [hunk 1]
```diff
@@ -1594,27 +1594,674 @@ var LibraryOpenAL = {
     return 0;
   },
 
+
```
Comment (kripken) [inline src/library_openal.js]: some unnecessary whitespace here

File: src/library_openal.js [hunk 2]
```diff
@@ -1594,27 +1594,674 @@ var LibraryOpenAL = {
...
-    Runtime.warnOnce('alSpeedOfSound() is not yet implemented! Ignoring all calls to it.');
+    Runtime.warnOnce('alSpeedOfSound() is not yet implemented!');
+    if (!AL.currentContext) {
+#if OPENAL_DEBUG
+      console.error("alSpeedOfSound called without a valid context");
+#endif
+      return;
+    }
+    if (value <= 0) { // Negative or zero values are disallowed
+        AL.currentContext.err = 0xA003 /* AL_INVALID_VALUE */;
```
Comment (kripken) [inline src/library_openal.js]: indentation looks too big here

File: src/library_openal.js [hunk 3]
```diff
@@ -1594,27 +1594,674 @@ var LibraryOpenAL = {
...
+#endif
+      return false;
+    }
+    // There's no defined capability for this yet - but extensions may want
+    // to use it.
+    AL.currentContext.err = 0xA002 /* AL_INVALID_ENUM */;
+    return false;
+  },
+
+
```
Comment (kripken) [inline src/library_openal.js]: unnecessary whitespace

File: src/library_openal.js [hunk 4]
```diff
@@ -1594,27 +1594,674 @@ var LibraryOpenAL = {
...
+    var val = AL.getDoubleHelper("alGetDoublev", param);
+    {{{ makeSetValue('data', '0', 'val', 'double') }}};
+  },
+
+  alGetFloat: function(param) {
+    return AL.getDoubleHelper("alGetFloat", param);
+  },
+
+  alGetFloatv: function(param, data) {
+    if (!data)
```
Comment (kripken) [inline src/library_openal.js]: if the if body is on another line, need `{,}`. But for things like this, can put the return on the same line

**Comment by kripken** [comment]
Thanks, nice work. I did a quick pass and had some style comments. Leaving to @juj for full review.

**Comment by jpernst** [comment]
Author of alto here. Funnily enough I've also started working on improving the AL implementation for the last week or so and was just about to ping the alto issue about it when I saw this; I wish I'd seen it sooner.

Perhaps we could combine efforts, since I'm more focused on a refactor of the implementation to fix some bugs and prepare for implementing some extensions. I don't know how cleanly my progress will rebase on top of this, but I'll explore that.

**Comment by yoanlcq** [comment]
@jpernst I agree! It would be a waste to somehow end up with two "concurrent" PRs.

Since my additions were fairly trivial (i.e not really "fixing" anything or adding new logic), I think it would be easiest to integrate them to your fork and close this PR in favor of an other, better one, when ready.
What would you say about this ? :)
(The timing is a bit unfortunate though since I won't be able to work on this for roughly a week starting from now, because exams etc. Well except perhaps for tiny, trivial changes, but that's it.)

**Comment by jpernst** [comment]
No problem, tomorrow I'll go through your PR and cherry pick everything you've added into mine and adapt it as necessary. There probably won't be much overlap, so it should hopefully go pretty smoothly.

**Comment by yoanlcq** [comment]
Ok - I'll close this then since this shouldn't get merged in the meantime.  

@jpernst would you mind adding me as a contributor to your fork so I can add my name to its AUTHORS file and otherwise contribute when I get time again ? Otherwise I'm fine with doing pull requests for this.

**Comment by jpernst** [comment]
Sure thing; I'll do that now.
